### PR TITLE
Deep sleep 2

### PR DESCRIPTION
**Pull request by martinwork**
Sibling PRs in codal-core, codal-nrf52, codal-microbit-v2

### CODE REVIEW

File: source/NRF52Serial.cpp [hunk 1]
```diff
@@ -296,3 +296,55 @@ void NRF52Serial::updateRxBufferAfterRXSTARTED()
...
+        if ( !(status & CODAL_SERIAL_STATUS_DEEPSLEEP))
+        {
+            if(txInUse() || rxInUse())
+                return DEVICE_SERIAL_IN_USE;
+
+            lockTx();
+            lockRx();
+
+            if(txBufferedSize() > 0)
+                disableInterrupt(TxInterrupt);
```
Comment (finneyj) [inline source/NRF52Serial.cpp:318]: Is this right @martinwork? Looks like we're intentionally disabled TX interrupts here when there's still data to send??

Comment (martinwork) [reply source/NRF52Serial.cpp:318]: Good point @finneyj! disableInterrupt(TxInterrupt) does nothing though.  See configurePins.

I'll swap it round so it makes more sense. Perhaps we do actually want to disable the interrupt here?

Comment (finneyj) [reply source/NRF52Serial.cpp:318]: Hmm... If it does nothing, do we need this bit of code at all?

Comment (martinwork) [reply source/NRF52Serial.cpp:318]: True!

**Comment by finneyj** [comment]
Thanks @martinwork!

**Pull request merged**